Sources/x10/swift_bindings/Device.swift

```diff
@@ -146,8 +150,49 @@ public struct Device {
 
   /// The current TF Eager device.
   public static var defaultTFEager: Device {
-    // TODO: Pull this from withDevice() {} mechanism?
-    return Device(kind: .CPU, ordinal: 0, backend: .TF_EAGER)
+    // Create a dummy tensor on any TFEager device.
+    var kind: Kind = .CPU
+    var ordinal = defaultOrdinal
+    let device = Device(kind: kind, ordinal: defaultOrdinal, backend: .TF_EAGER)
+    let tensor = Tensor<Float>(zeros: [1], on: device)
+    let handle = tensor.handle._cTensorHandle
+    let status = TF_NewStatus()
+
+    // Find out what the underlying libraries think the default is.
+    if let cString = TFE_TensorHandleDeviceName(handle, status) {
+      checkOk(status)
+
+      // TODO: What's the best way to deallocate this memory?
+      // defer { DeleteString(str) }
+      let tfDeviceName = String(cString: cString)
+
+      // Parse type and ordinal from a string with the expected syntax:
+      //   /job:localhost/replica:0/task:0/device:CPU:0
+      let pattern = ".+device:(.+):(\\d+)$"
+      let regex = try! NSRegularExpression(pattern: pattern)
```

Comment:
This string parsing looks expensive. Might want to double check that this doesn't happen too much.